dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -383,15 +383,19 @@ HRESULT WINAPI CRecycleBinItemContextMenu::InvokeCommand(LPCMINVOKECOMMANDINFO l
         if (lpcmi->lpVerb == MAKEINTRESOURCEA(1))
         {
             /* restore file */
-            if (RestoreFile(Context.hDeletedFile))
-                return S_OK;
-            else
-                return E_FAIL;
+            BOOL ret = RestoreFile(Context.hDeletedFile);
+            
+            CloseRecycleBinHandle(Context.hDeletedFile);
+            
+            return (ret) ? S_OK : E_FAIL;
         }
         else
         {
             DeleteFileHandleToRecycleBin(Context.hDeletedFile);
-            return E_NOTIMPL;
```

Comment:
Actually, what does this case correspond to?

Reply:
well, actually this is implemented, idk why originally returns that this isn't implemented